app/controllers/api/v1/site_settings_controller.rb

```diff
@@ -3,10 +3,27 @@
 module Api
   module V1
     class SiteSettingsController < ApiController
+      def index
+        site_setting_data = {}
+        site_settings = SiteSetting.joins(:setting).select(:id, :name, :value)
+
+        site_settings.each do |setting|
+          site_setting_data[setting.name] = { value: setting.value, id: setting.id }
+        end
+
+        render_json data: site_setting_data, status: :ok
+      end
+
       # GET /api/v1/site_settings/:name
       def show
         render_json data: SettingGetter.new(setting_name: params[:name], provider: 'greenlight').call # TODO: - ahmad: fix provider
       end
+
+      def update
+        SiteSetting.find_by(setting_id: params[:settingId]).update(value: params[:settingValue].to_s)
+
+        render_data status: :ok
```

Comment:
Since sometimes we'd be using the `render_json` then IMO we should be using `render_data` only when requiring the formatting by one of our serializers.